monai/apps/reconstruction/complex_utils.py

```diff
@@ -0,0 +1,182 @@
+            )
+            return converted_data
+    else:
+        if not np.iscomplexobj(data):
+            converted_data = convert_to_tensor(
+                data, dtype=dtype, device=device, wrap_sequence=wrap_sequence, track_meta=track_meta
+            )
+            return converted_data
+
+    # if data is complex, turn its stacked version into a tensor
+    if isinstance(data, torch.Tensor):
+        data = torch.stack([data.real, data.imag], dim=-1)
+
+    elif isinstance(data, np.ndarray):
+        if re.search(r"[SaUO]", data.dtype.str) is None:
+            # numpy array with 0 dims is also sequence iterable,
+            # `ascontiguousarray` will add 1 dim if img has no dim, so we only apply on data with dims
+            if data.ndim > 0:
+                data = np.ascontiguousarray(data)
+            data = np.stack((data.real, data.imag), axis=-1)
+
+    elif isinstance(data, (float, int)):
+        data = [[data.real, data.imag]]
+
+    elif isinstance(data, list):
+        data = convert_to_numpy(data, wrap_sequence=True)
+        data = np.stack((data.real, data.imag), axis=-1).tolist()
+
+    converted_data = convert_to_tensor(
+        data, dtype=dtype, device=device, wrap_sequence=wrap_sequence, track_meta=track_meta
+    )
+    return converted_data
+
+
+def complex_abs(x: NdarrayOrTensor) -> NdarrayOrTensor:
+    """
+    Compute the absolute value of a complex array.
+
+    Args:
+        x: Input array/tensor with 2 channels in the last dimension representing real and imaginary parts.
+
+    Returns:
+        Absolute value along the last dimention
+
+    Example:
+        .. code-block:: python
+
+            import numpy as np
+            x = np.array([3,4])[np.newaxis]
+            # the following line prints 5
+            print(complex_abs(x))
+    """
+    if x.shape[-1] != 2:
+        raise ValueError(f"x.shape[-1] is not 2 ({x.shape[-1]}).")
+    return (x[..., 0] ** 2 + x[..., 1] ** 2) ** 0.5
+
+
+def complex_mul(x: NdarrayOrTensor, y: NdarrayOrTensor) -> NdarrayOrTensor:
+    """
+    Compute complex-valued multiplication. Supports Ndim inputs with last dim equal to 2 (real/imaginary channels)
+
+    Args:
+        x: Input array/tensor with 2 channels in the last dimension representing real and imaginary parts.
+        y: Input array/tensor with 2 channels in the last dimension representing real and imaginary parts.
+
+    Returns:
+        Complex multiplication of x and y
+
+    Example:
+        .. code-block:: python
+
+            import numpy as np
+            x = np.array([[1,2],[3,4]])
+            y = np.array([[1,1],[1,1]])
+            # the following line prints array([[-1,  3], [-1,  7]])
+            print(complex_mul(x,y))
+    """
+    if x.shape[-1] != 2 or y.shape[-1] != 2:
+        raise ValueError(f"last dim must be 2, but x.shape[-1] is {x.shape[-1]} and y.shape[-1] is {y.shape[-1]}.")
+
+    re = x[..., 0] * y[..., 0] - x[..., 1] * y[..., 1]
```

Comment:
there's a tiny issue with the variable name `re` happens to be redefining the `import re` at line 15, perhaps we can rectify this later...